cohere/responses/custom_model.py

```diff
@@ -41,6 +42,41 @@
 }
 
 
+@dataclass
+class HyperParameters:
+    early_stopping_patience: int
+    early_stopping_threshold: float
+    train_batch_size: int
+    train_steps: int
+    learning_rate: float
+
+    @staticmethod
+    def from_response(response: Optional[dict]) -> "HyperParameters":
+        return HyperParameters(
+            early_stopping_patience=response["earlyStoppingPatience"],
+            early_stopping_threshold=response["earlyStoppingThreshold"],
+            train_batch_size=response["trainBatchSize"],
+            train_steps=response["trainSteps"],
+            learning_rate=response["learningRate"],
+        )
+
+
+class HyperParametersInput(TypedDict):
+    """
+    early_stopping_patience: int (default=6, min=0, max=10)
+    early_stopping_threshold: float (default=0.01, min=0, max=0.1)
+    train_batch_size: int (default=16, min=2, max=16)
+    train_steps: int (default=2500, min=100, max=20000)
+    learning_rate: float (default=0.01, min=0.000005, max=0.1)
```

Comment:
Just curious - where are these defaults/limits enforced?

Reply:
They are enforced in the backend:

https://github.com/cohere-ai/blobheart/pull/10064/files#diff-1727e182d1b97ea0c14b8a15c5f77c581ba6c508a062aa4a241c75778a511836R17
AND
https://github.com/cohere-ai/blobheart/pull/10064/files#diff-1727e182d1b97ea0c14b8a15c5f77c581ba6c508a062aa4a241c75778a511836R45